plugins/generator-1.18.2/forge-1.18.2/templates/block/block.java.ftl

```diff
@@ -259,44 +269,58 @@ public class ${name}Block extends
 		}
 		</#if>
 
+	<#if data.rotationMode == 1 && data.enablePitch>
+	public AttachFace faceForDirection(Direction direction) {
+		if (direction.getAxis() == Direction.Axis.Y)
+			return direction == Direction.UP ? AttachFace.CEILING : AttachFace.FLOOR;
+		else
+			return AttachFace.WALL;
+	}
+	</#if>
+
 	@Override
 	public BlockState getStateForPlacement(BlockPlaceContext context) {
-	    <#if data.rotationMode == 4>
-	    Direction facing = context.getClickedFace();
-	    </#if>
-	    <#if data.rotationMode == 5>
-	    Direction.Axis axis = context.getClickedFace().getAxis();
-	    </#if>
-	    <#if data.isWaterloggable>
-	    boolean flag = context.getLevel().getFluidState(context.getClickedPos()).getType() == Fluids.WATER;
-	    </#if>;
+		<#if data.isWaterloggable>
+		boolean flag = context.getLevel().getFluidState(context.getClickedPos()).getType() == Fluids.WATER;
+		</#if>
 		<#if data.rotationMode != 3>
 		return this.defaultBlockState()
 		        <#if data.rotationMode == 1>
+		            <#if data.enablePitch>
+		            .setValue(FACE, faceForDirection(context.getNearestLookingDirection().getOpposite()))
+		            </#if>
 		        .setValue(FACING, context.getHorizontalDirection().getOpposite())
 		        <#elseif data.rotationMode == 2>
 		        .setValue(FACING, context.getNearestLookingDirection().getOpposite())
 	            <#elseif data.rotationMode == 4>
-		        .setValue(FACING, facing)
+		        .setValue(FACING, context.getClickedFace())
 	            <#elseif data.rotationMode == 5>
-	            .setValue(AXIS, axis)
+	            .setValue(AXIS, context.getClickedFace().getAxis())
 		        </#if>
 		        <#if data.isWaterloggable>
 		        .setValue(WATERLOGGED, flag)
-		        </#if>
+		        </#if>;
 		<#elseif data.rotationMode == 3>
-	    if (context.getClickedFace() == Direction.UP || context.getClickedFace() == Direction.DOWN)
+	    if (context.getClickedFace().getAxis() == Direction.Axis.Y)
 	        return this.defaultBlockState()
-	                .setValue(FACING, Direction.NORTH)
+	                <#if data.enablePitch>
+	                    .setValue(FACE, context.getClickedFace().getOpposite() == Direction.UP ? AttachFace.CEILING : AttachFace.FLOOR)
+	                    .setValue(FACING, context.getHorizontalDirection())
+	                <#else>
+	                    .setValue(FACING, Direction.NORTH)
+	                </#if>
 	                <#if data.isWaterloggable>
 	                .setValue(WATERLOGGED, flag)
 	                </#if>;
 	    return this.defaultBlockState()
-	            .setValue(FACING, context.getClickedFace())
+	            <#if data.enablePitch>
+	                .setValue(FACE, AttachFace.WALL)
+	            </#if>
+	            .setValue(FACING, context.getClickedFace().getOpposite())

```

Comment:
You should remove the `.getOpposite()`, the blocks are being placed in the wrong direction now